fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java

```diff
@@ -113,13 +114,33 @@ public void analyze(StatementBase statement, ConnectContext session) {
         public Void visitAnalyzeStatement(AnalyzeStmt statement, ConnectContext session) {
             statement.getTableName().normalization(session);
             Table analyzeTable = MetaUtils.getSessionAwareTable(session, null, statement.getTableName());
-
+            AnalyzeTypeDesc analyzeTypeDesc = statement.getAnalyzeTypeDesc();
             if (StatisticUtils.statisticDatabaseBlackListCheck(statement.getTableName().getDb())) {
                 throw new SemanticException("Forbidden collect database: %s", statement.getTableName().getDb());
             }
 
             // ANALYZE TABLE xxx (col1, col2, ...)
             List<Expr> columns = statement.getColumns();
+
+            if (analyzeTypeDesc instanceof AnalyzeMultiColumnDesc) {
+                if (columns.size() <= 1) {
+                    throw new SemanticException("must greater than 1 column on multi-column combined analyze statement");
+                }
+
+                if (columns.size() > Config.statistics_max_multi_column_combined_num) {
+                    throw new SemanticException("column size " + columns.size() + " exceeded max size of " +
+                            Config.statistics_max_multi_column_combined_num + " on multi-column combined analyze statement");
+                }
+
+                if (statement.getPartitionNames() != null) {
+                    throw new SemanticException("not support specify partition names on multi-column analyze statement");
+                }
+
+                if (statement.isAsync()) {
+                    throw new SemanticException("not support async analyze on multi-column analyze statement");
```

Comment:
why ?

Reply:
Temporarily consistent with the histogram